src/main/java/com/openlattice/launchpad/configuration/LaunchpadDestination.java

```diff
@@ -26,74 +26,125 @@
+
+    private static final Logger      logger = LoggerFactory.getLogger( LaunchpadDestination.class );
+
+    private final String        name;
+    private final String        username;
+    private final String        password;
+    private final String        writeUrl;
+    private final String        writeDriver;
+    private final String        dataFormat;
+    private final Properties    properties;
+    private final int           batchSize;
+    private final SaveMode      writeMode;
 
     public LaunchpadDestination(
             @JsonProperty( NAME ) String name,
-            @JsonProperty( WRITE_URL ) String writeUrl,
-            @JsonProperty( WRITE_DRIVER ) String writeDriver,
+            @JsonProperty( URL ) String writeUrl,
+            @JsonProperty( DRIVER ) String writeDriver,
+            @JsonProperty( DATA_FORMAT ) Optional<String> dataFormat,
             @JsonProperty( WRITE_MODE ) Optional<String> writeMode,
-            @JsonProperty( USER ) Optional<String> username,
+            @JsonProperty( USERNAME ) Optional<String> username,
             @JsonProperty( PASSWORD ) Optional<String> password,
             @JsonProperty( PROPERTIES ) Optional<Properties> properties,
             @JsonProperty( BATCH_SIZE ) Optional<Integer> batchSize ) {
         Preconditions.checkState( StringUtils.isNotBlank( name ), "Name must be specified for a desintation." );
         this.name = name;
         this.writeUrl = writeUrl;
         this.writeDriver = writeDriver;
-        this.batchSize = batchSize.orElse( DEFAULT_BATCH_SIZE );
+        this.batchSize = batchSize.orElse( DEFAULT_DATA_CHUNK_SIZE );
+        // if dataFormat not set, use writeDriver value
+        this.dataFormat = dataFormat.orElse( writeDriver );
         this.properties = properties.orElse( new Properties() );
         this.writeMode = SaveMode.valueOf( writeMode.orElse( DEFAULT_WRITE_MODE.name()));
 
-        this.properties.put(JDBC_URL, writeUrl);
-        this.properties.put(MAXIMUM_POOL_SIZE, "1");
-        this.properties.put(CONNECTION_TIMEOUT, "120000"); //2-minute connection timeout
-        username.ifPresent( u -> this.properties.setProperty( "user", u ) );
-        username.ifPresent( u -> this.properties.setProperty( "username", u ) );
-        password.ifPresent( p -> this.properties.setProperty( PASSWORD, p ) );
+        this.username = username.orElse( "" );
+        this.password = password.orElse( "" );
+
+        // JDBC datasource
+        if ( !NON_JDBC_DRIVERS.contains( writeDriver ) ){
+            if ( StringUtils.isBlank( this.password )){
+                logger.warn( "connecting to " + name + " with blank password!");
+            }
+            this.properties.put(JDBC_URL, writeUrl);
+            this.properties.put(MAXIMUM_POOL_SIZE, "1");
+            this.properties.put(CONNECTION_TIMEOUT, "120000"); //2-minute connection timeout
+            this.properties.setProperty( USER, this.username );
```

Comment:
So I hadn't looked into this before but actually "user" is the name of the property used by Oracle, Postgres, pervasive and IBM at the least. I think the intent originally was to provide both "user" and "username" just in case we're using some funky driver